TestNotifyClosed: bind: address already in use

The test suite frequently fails using v1.2.4 with Go 1.6 on Linux x86_64 with the following output:
```
go test -v gopkg.in/tylerb/graceful.v1
=== RUN   TestGracefulRun
--- PASS: TestGracefulRun (1.25s)
=== RUN   TestGracefulRunTimesOut
--- PASS: TestGracefulRunTimesOut (2.60s)
=== RUN   TestGracefulRunDoesntTimeOut
--- PASS: TestGracefulRunDoesntTimeOut (2.00s)
=== RUN   TestGracefulRunNoRequests
--- PASS: TestGracefulRunNoRequests (1.00s)
=== RUN   TestGracefulForwardsConnState
--- PASS: TestGracefulForwardsConnState (2.25s)
=== RUN   TestGracefulExplicitStop
--- PASS: TestGracefulExplicitStop (1.10s)
=== RUN   TestGracefulExplicitStopOverride
--- PASS: TestGracefulExplicitStopOverride (1.10s)
=== RUN   TestBeforeShutdownAndShutdownInitiatedCallbacks
--- PASS: TestBeforeShutdownAndShutdownInitiatedCallbacks (1.10s)
=== RUN   TestNotifyClosed
--- FAIL: TestNotifyClosed (0.00s)
        graceful_test.go:362: listen tcp :9654: bind: address already in use
=== RUN   TestStopDeadlock
--- PASS: TestStopDeadlock (1.00s)
=== RUN   TestStopRace
--- PASS: TestStopRace (1.00s)
FAIL
exit status 1
FAIL    gopkg.in/tylerb/graceful.v1     14.415s
```

Another failure using the same test environment:
```
        go test -v -run ^Test\(GracefulRun\[A-Z\]\|Before\|Stop\) gopkg.in/tylerb/graceful.v1
=== RUN   TestGracefulRunTimesOut
--- PASS: TestGracefulRunTimesOut (2.60s)
=== RUN   TestGracefulRunDoesntTimeOut
--- FAIL: TestGracefulRunDoesntTimeOut (1.20s)
        graceful_test.go:44: No response when a response was expected.
=== RUN   TestGracefulRunNoRequests
--- PASS: TestGracefulRunNoRequests (1.00s)
=== RUN   TestBeforeShutdownAndShutdownInitiatedCallbacks
--- PASS: TestBeforeShutdownAndShutdownInitiatedCallbacks (1.10s)
=== RUN   TestStopDeadlock
--- PASS: TestStopDeadlock (1.00s)
=== RUN   TestStopRace
--- PASS: TestStopRace (1.00s)
FAIL
exit status 1
FAIL    gopkg.in/tylerb/graceful.v1     7.910s
```

This looks like a race condition. Do you think it’s in the tests or the package itself?

--- a/graceful_test.go
+++ b/graceful_test.go
@@ -290,6 +290,9 @@ func TestGracefulExplicitStopOverride(t *testing.T) {
 }
 
 func TestBeforeShutdownAndShutdownInitiatedCallbacks(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
 	server, l, err := createListener(1 * time.Millisecond)
 	if err != nil {
 		t.Fatal(err)
@@ -301,9 +304,11 @@ func TestBeforeShutdownAndShutdownInitiatedCallbacks(t *testing.T) {
 	cb2 := func() { close(shutdownInitiatedCalled) }
 
 	srv := &Server{Server: server, BeforeShutdown: cb1, ShutdownInitiated: cb2}
-
 	go func() {
-		go srv.Serve(l)
+		srv.Serve(l)
+		wg.Done()
+	}()
+	go func() {
 		time.Sleep(waitTime)
 		srv.Stop(killTime)
 	}()
@@ -329,7 +334,10 @@ func TestBeforeShutdownAndShutdownInitiatedCallbacks(t *testing.T) {
 	if !shutdownInitiated {
 		t.Fatal("shutdownInitiated should be true")
 	}
+
+	wg.Wait()
 }
+
 func hijackingListener(srv *Server) (*http.Server, net.Listener, error) {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", func(rw http.ResponseWriter, r *http.Request) {

